compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -632,21 +632,46 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
 
                 if self.is_fn_ty(rcvr_ty, span) {
-                    fn report_function<T: std::fmt::Display>(
-                        err: &mut DiagnosticBuilder<'_>,
-                        name: T,
-                    ) {
-                        err.note(
-                            &format!("`{}` is a function, perhaps you wish to call it", name,),
-                        );
-                    }
-
                     if let SelfSource::MethodCall(expr) = source {
-                        if let Ok(expr_string) = tcx.sess.source_map().span_to_snippet(expr.span) {
-                            report_function(&mut err, expr_string);
+                        let snippet: Option<String> = if let Ok(expr_string) =
+                            tcx.sess.source_map().span_to_snippet(expr.span)
+                        {
+                            Some(expr_string)
                         } else if let ExprKind::Path(QPath::Resolved(_, path)) = expr.kind {
                             if let Some(segment) = path.segments.last() {
-                                report_function(&mut err, segment.ident);
+                                Some(segment.ident.to_string())
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        };
+
+                        if let Some(snippet) = snippet {
+                            let node = if let ty::FnDef(def_id, _) = rcvr_ty.kind() {
+                                let hir_id =
+                                    tcx.hir().local_def_id_to_hir_id(def_id.expect_local());
+                                let node = tcx.hir().get(hir_id);
+                                Some(node)
+                            } else {
+                                None
+                            };
+
+                            // If the function is an empty-tuple constructor
+                            // we recommend that they call it
+                            if let Some(Node::Ctor(&hir::VariantData::Tuple(&[], _))) = node {
+                                err.help(&format!("`{}` is the constructor of an empty-tuple struct or enum variant. Consider calling it?", snippet));
```

Comment:
This is redundant since the suggestion says "call the constructor".

```suggestion
                                err.help(&format!("`{}` is the constructor of an empty-tuple struct or enum variant", snippet));
```